In [1]:
import numpy as np
import pandas as pd
import plotly.express as px

In [2]:
pd.set_option('display.max_columns', None)

In [3]:
df=pd.read_csv('master.csv')
df

,country,year,sex,age,suicides_no,population,suicides/100k pop,country-year,HDI for year,gdp_for_year ($),gdp_per_capita ($),generation
0,Albania,1987,male,15-24 years,21,312900,6.71,Albania1987,NaN,"2,156,624,900",796,Generation X
1,Albania,1987,male,35-54 years,16,308000,5.19,Albania1987,NaN,"2,156,624,900",796,Silent
2,Albania,1987,female,15-24 years,14,289700,4.83,Albania1987,NaN,"2,156,624,900",796,Generation X
3,Albania,1987,male,75+ years,1,21800,4.59,Albania1987,NaN,"2,156,624,900",796,G.I. Generation
4,Albania,1987,male,25-34 years,9,274300,3.28,Albania1987,NaN,"2,156,624,900",796,Boomers
...,...,...,...,...,...,...,...,...,...,...,...,...
27815,Uzbekistan,2014,female,35-54 years,107,3620833,2.96,Uzbekistan2014,0.675,"63,067,077,179",2309,Generation X
27816,Uzbekistan,2014,female,75+ years,9,348465,2.58,Uzbekistan2014,0.675,"63,067,077,179",2309,Silent
27817,Uzbekistan,2014,male,5-14 years,60,2762158,2.17,Uzbekistan2014,0.675,"63,067,077,179",2309,Generation Z
27818,Uzbekistan,2014,female,5-14 years,44,2631600,1.67,Uzbekistan2014,0.675,"63,067,077,179",2309,Generation Z


# Data Understanding

### Dataset Column Descriptions

| Column Name | Description |
| :--- | :--- |
| **country** | The name of the country where the data was recorded. |
| **year** | The specific year of the recorded data. |
| **sex** | The biological sex of the target group (Male or Female). |
| **age** | The specific age range (e.g., 15-24 years, 35-54 years). |
| **suicides_no** | The total number of recorded suicides for that specific group. |
| **population** | The total population count for that specific demographic group. |
| **suicides/100k pop** | The suicide rate calculated per 100,000 inhabitants. |
| **country-year** | A unique identifier combining country name and year. |
| **HDI for year** | Human Development Index for the country in that specific year. |
| **gdp_for_year ($)** | The total Gross Domestic Product of the country for that year. |
| **gdp_per_capita ($)** | The GDP divided by the total population (Average income indicator). |
| **generation** | The generation name based on the age group (e.g., Boomers, Gen X, Millennials). |

## Data Exploration

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 27820 entries, 0 to 27819
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   country             27820 non-null  str    
 1   year                27820 non-null  int64  
 2   sex                 27820 non-null  str    
 3   age                 27820 non-null  str    
 4   suicides_no         27820 non-null  int64  
 5   population          27820 non-null  int64  
 6   suicides/100k pop   27820 non-null  float64
 7   country-year        27820 non-null  str    
 8   HDI for year        8364 non-null   float64
 9    gdp_for_year ($)   27820 non-null  str    
 10  gdp_per_capita ($)  27820 non-null  int64  
 11  generation          27820 non-null  str    
dtypes: float64(2), int64(4), str(6)
memory usage: 4.2 MB


# statistics Summary 

In [5]:
df.describe().round(2)

,year,suicides_no,population,suicides/100k pop,HDI for year,gdp_per_capita ($)
count,27820.00,27820.00,27820.00,27820.00,8364.00,27820.00
mean,2001.26,242.57,1844793.62,12.82,0.78,16866.46
std,8.47,902.05,3911779.44,18.96,0.09,18887.58
min,1985.00,0.00,278.00,0.00,0.48,251.00
25%,1995.00,3.00,97498.50,0.92,0.71,3447.00
50%,2002.00,25.00,430150.00,5.99,0.78,9372.00
75%,2008.00,131.00,1486143.25,16.62,0.86,24874.00
max,2016.00,22338.00,43805214.00,224.97,0.94,126352.00


In [6]:
num_col=df.select_dtypes(include=['int64', 'float64']).columns
num_col

Index(['year', 'suicides_no', 'population', 'suicides/100k pop',
       'HDI for year', 'gdp_per_capita ($)'],
      dtype='str')

In [7]:
# boxplot for all num_col

for col in num_col:
    px.box(df, x=col).show()

### there is skew in Data beacause of outliers , so i will use median in my calculation instead of mean

# Categorical Summary

In [8]:
df.describe(include='object').round(2)

C:\Users\HP\AppData\Local\Temp\ipykernel_20188\1379714876.py:1: Pandas4Warning:

For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.



,country,sex,age,country-year,gdp_for_year ($),generation
count,27820,27820,27820,27820,27820,27820
unique,101,2,6,2321,2321,6
top,Austria,male,15-24 years,Albania1987,"2,156,624,900",Generation X
freq,382,13910,4642,12,12,6408


# Delete unused Column

In [9]:
df.drop(labels=['country-year'],axis=1,inplace=True)

In [10]:
df.columns

Index(['country', 'year', 'sex', 'age', 'suicides_no', 'population',
       'suicides/100k pop', 'HDI for year', ' gdp_for_year ($) ',
       'gdp_per_capita ($)', 'generation'],
      dtype='str')

# check for duplicates

In [11]:
df.duplicated().sum()

np.int64(0)

# Check for Nulls

In [12]:
(df.isna().mean()*100).round(2)

country                0.00
year                   0.00
sex                    0.00
age                    0.00
suicides_no            0.00
population             0.00
suicides/100k pop      0.00
HDI for year          69.94
 gdp_for_year ($)      0.00
gdp_per_capita ($)     0.00
generation             0.00
dtype: float64

# Nulls >50% => Drop the column

In [13]:
df.drop(labels=['HDI for year'],axis=1,inplace=True)

In [14]:
df.columns

Index(['country', 'year', 'sex', 'age', 'suicides_no', 'population',
       'suicides/100k pop', ' gdp_for_year ($) ', 'gdp_per_capita ($)',
       'generation'],
      dtype='str')

In [15]:
(df.isna().mean()*100).round(2)

country               0.0
year                  0.0
sex                   0.0
age                   0.0
suicides_no           0.0
population            0.0
suicides/100k pop     0.0
 gdp_for_year ($)     0.0
gdp_per_capita ($)    0.0
generation            0.0
dtype: float64

# Feature Engineering(Handle the Problem of Age column)

In [16]:
df['age'].unique()

<ArrowStringArray>
['15-24 years', '35-54 years',   '75+ years', '25-34 years', '55-74 years',
  '5-14 years']
Length: 6, dtype: str

In [17]:
## Add a new column to refer for each cell with a number
clean_ages = {
    '5-14 years': 0,
    '15-24 years': 1,
    '25-34 years': 2,
    '35-54 years': 3,
    '55-74 years': 4,
    '75+ years': 5
}

df['new_age'] = df['age'].replace(clean_ages).astype(int)

In [18]:
df[['age', 'new_age']].head()

,age,new_age
0,15-24 years,1
1,35-54 years,3
2,15-24 years,1
3,75+ years,5
4,25-34 years,2


# clean columns names

In [19]:
df.columns

Index(['country', 'year', 'sex', 'age', 'suicides_no', 'population',
       'suicides/100k pop', ' gdp_for_year ($) ', 'gdp_per_capita ($)',
       'generation', 'new_age'],
      dtype='str')

In [20]:
df.columns=df.columns.str.strip()

### refer to male=1 and female=0 to make analysis more efficient and easy

In [21]:
sex_mapping = {'male': 1, 'female': 0}

df['sex_numeric'] = df['sex'].replace(sex_mapping).astype(int)

In [22]:
df.head()

,country,year,sex,age,suicides_no,population,suicides/100k pop,gdp_for_year ($),gdp_per_capita ($),generation,new_age,sex_numeric
0,Albania,1987,male,15-24 years,21,312900,6.71,"2,156,624,900",796,Generation X,1,1
1,Albania,1987,male,35-54 years,16,308000,5.19,"2,156,624,900",796,Silent,3,1
2,Albania,1987,female,15-24 years,14,289700,4.83,"2,156,624,900",796,Generation X,1,0
3,Albania,1987,male,75+ years,1,21800,4.59,"2,156,624,900",796,G.I. Generation,5,1
4,Albania,1987,male,25-34 years,9,274300,3.28,"2,156,624,900",796,Boomers,2,1


In [23]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 27820 entries, 0 to 27819
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   country             27820 non-null  str    
 1   year                27820 non-null  int64  
 2   sex                 27820 non-null  str    
 3   age                 27820 non-null  str    
 4   suicides_no         27820 non-null  int64  
 5   population          27820 non-null  int64  
 6   suicides/100k pop   27820 non-null  float64
 7   gdp_for_year ($)    27820 non-null  str    
 8   gdp_per_capita ($)  27820 non-null  int64  
 9   generation          27820 non-null  str    
 10  new_age             27820 non-null  int64  
 11  sex_numeric         27820 non-null  int64  
dtypes: float64(1), int64(6), str(5)
memory usage: 3.8 MB


### In Depth Check for Categorical columns

In [24]:
cat_cols = df.select_dtypes(include= 'str').columns
cat_cols

Index(['country', 'sex', 'age', 'gdp_for_year ($)', 'generation'], dtype='str')

### Checking unique country names to ensure there are no duplicates or inconsistent naming (e.g., USA vs United States).

In [25]:
sorted(df['country'].unique())


['Albania',
 'Antigua and Barbuda',
 'Argentina',
 'Armenia',
 'Aruba',
 'Australia',
 'Austria',
 'Azerbaijan',
 'Bahamas',
 'Bahrain',
 'Barbados',
 'Belarus',
 'Belgium',
 'Belize',
 'Bosnia and Herzegovina',
 'Brazil',
 'Bulgaria',
 'Cabo Verde',
 'Canada',
 'Chile',
 'Colombia',
 'Costa Rica',
 'Croatia',
 'Cuba',
 'Cyprus',
 'Czech Republic',
 'Denmark',
 'Dominica',
 'Ecuador',
 'El Salvador',
 'Estonia',
 'Fiji',
 'Finland',
 'France',
 'Georgia',
 'Germany',
 'Greece',
 'Grenada',
 'Guatemala',
 'Guyana',
 'Hungary',
 'Iceland',
 'Ireland',
 'Israel',
 'Italy',
 'Jamaica',
 'Japan',
 'Kazakhstan',
 'Kiribati',
 'Kuwait',
 'Kyrgyzstan',
 'Latvia',
 'Lithuania',
 'Luxembourg',
 'Macau',
 'Maldives',
 'Malta',
 'Mauritius',
 'Mexico',
 'Mongolia',
 'Montenegro',
 'Netherlands',
 'New Zealand',
 'Nicaragua',
 'Norway',
 'Oman',
 'Panama',
 'Paraguay',
 'Philippines',
 'Poland',
 'Portugal',
 'Puerto Rico',
 'Qatar',
 'Republic of Korea',
 'Romania',
 'Russian Federation',
 'Saint 

In [26]:
for col in cat_cols:
    print(col)
    print(df[col].nunique())
    print(df[col].unique())
    print('-' * 100)

country
101
<ArrowStringArray>
[             'Albania',  'Antigua and Barbuda',            'Argentina',
              'Armenia',                'Aruba',            'Australia',
              'Austria',           'Azerbaijan',              'Bahamas',
              'Bahrain',
 ...
             'Thailand',  'Trinidad and Tobago',               'Turkey',
         'Turkmenistan',              'Ukraine', 'United Arab Emirates',
       'United Kingdom',        'United States',              'Uruguay',
           'Uzbekistan']
Length: 101, dtype: str
----------------------------------------------------------------------------------------------------
sex
2
<ArrowStringArray>
['male', 'female']
Length: 2, dtype: str
----------------------------------------------------------------------------------------------------
age
6
<ArrowStringArray>
['15-24 years', '35-54 years',   '75+ years', '25-34 years', '55-74 years',
  '5-14 years']
Length: 6, dtype: str
---------------------------------------------

In [27]:
df['generation']=df['generation'].replace('Millenials','Millennials')

In [28]:
df['generation'] = df['generation'].str.strip()

### Converted gdp_for_year ($) from string to float by removing commas to allow for quantitative analysis and correlation with suicide rates.

In [29]:
df['gdp_for_year ($)'] =df['gdp_for_year ($)'].str.replace(',', '').astype(float)
    
df['gdp_for_year ($)'] 

0        2.156625e+09
1        2.156625e+09
2        2.156625e+09
3        2.156625e+09
4        2.156625e+09
             ...     
27815    6.306708e+10
27816    6.306708e+10
27817    6.306708e+10
27818    6.306708e+10
27819    6.306708e+10
Name: gdp_for_year ($), Length: 27820, dtype: float64

In [30]:
df.duplicated().sum()

np.int64(0)

## Created a new column 'gdp_billion' by scaling down the total GDP values. This simplifies the data for better readability and more intuitive visualization.

In [31]:
df['gdp_billion'] = df['gdp_for_year ($)'] / 1e9

##### Defined a custom logical order for the 'generation' column using pd.Categorical. This ensures that all future plots and analyses follow the correct chronological sequence of generations rather than alphabetical order.

In [ ]:
generation_order = [
    'G.I. Generation', 
    'Silent', 
    'Boomers', 
    'Generation X', 
    'Millennials', 
    'Generation Z'
]

df['generation'] = pd.Categorical(df['generation'], categories=generation_order, ordered=True)



In [33]:
df.columns

Index(['country', 'year', 'sex', 'age', 'suicides_no', 'population',
       'suicides/100k pop', 'gdp_for_year ($)', 'gdp_per_capita ($)',
       'generation', 'new_age', 'sex_numeric', 'gdp_billion'],
      dtype='str')

In [34]:
df.duplicated().sum()

np.int64(0)

In [35]:
df.isna().sum()

country               0
year                  0
sex                   0
age                   0
suicides_no           0
population            0
suicides/100k pop     0
gdp_for_year ($)      0
gdp_per_capita ($)    0
generation            0
new_age               0
sex_numeric           0
gdp_billion           0
dtype: int64

### In Depth check for numerical columns

In [36]:
num_cols = df.select_dtypes(include= 'number').columns.str.strip()
num_cols

Index(['year', 'suicides_no', 'population', 'suicides/100k pop',
       'gdp_for_year ($)', 'gdp_per_capita ($)', 'new_age', 'sex_numeric',
       'gdp_billion'],
      dtype='str')

In [37]:
for num in num_cols:
    print(num)
    print(df[num].nunique())
    print(df[num].unique())
    print('-' * 100)

year
32
[1987 1988 1989 1992 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002
 2003 2004 2005 2006 2007 2008 2009 2010 1985 1986 1990 1991 2012 2013
 2014 2015 2011 2016]
----------------------------------------------------------------------------------------------------
suicides_no
2084
[  21   16   14 ... 5503 4359 2872]
----------------------------------------------------------------------------------------------------
population
25564
[ 312900  308000  289700 ... 2762158 2631600 1438935]
----------------------------------------------------------------------------------------------------
suicides/100k pop
5298
[ 6.71  5.19  4.83 ... 47.86 40.75 26.61]
----------------------------------------------------------------------------------------------------
gdp_for_year ($)
2321
[2.15662490e+09 2.12600000e+09 2.33512499e+09 ... 5.18215733e+10
 5.76904535e+10 6.30670772e+10]
----------------------------------------------------------------------------------------------------
gdp_per_capita 

In [38]:
df.columns

Index(['country', 'year', 'sex', 'age', 'suicides_no', 'population',
       'suicides/100k pop', 'gdp_for_year ($)', 'gdp_per_capita ($)',
       'generation', 'new_age', 'sex_numeric', 'gdp_billion'],
      dtype='str')

## EDA For Answering Questions

In [43]:
correlation_cols = ['suicides_no', 'population', 'suicides/100k pop', 'gdp_for_year ($)',
                    'gdp_per_capita ($)', 'new_age', 'sex_numeric']

correlation_matrix = df[correlation_cols].corr()
print(correlation_matrix)

px.imshow(correlation_matrix, text_auto=True, aspect='auto').show()

                    suicides_no  population  suicides/100k pop  \
suicides_no            1.000000    0.616162           0.306604   
population             0.616162    1.000000           0.008285   
suicides/100k pop      0.306604    0.008285           1.000000   
gdp_for_year ($)       0.430096    0.710697           0.025240   
gdp_per_capita ($)     0.061330    0.081510           0.001785   
new_age                0.079892   -0.060716           0.362587   
sex_numeric            0.144629   -0.011242           0.391496   

                    gdp_for_year ($)  gdp_per_capita ($)       new_age  \
suicides_no             4.300959e-01        6.132975e-02  7.989236e-02   
population              7.106973e-01        8.150986e-02 -6.071612e-02   
suicides/100k pop       2.523964e-02        1.785134e-03  3.625866e-01   
gdp_for_year ($)        1.000000e+00        3.034045e-01 -3.045700e-04   
gdp_per_capita ($)      3.034045e-01        1.000000e+00  7.849172e-04   
new_age                -3.0

# 1. Demographic Analysis

In [44]:
df.columns

Index(['country', 'year', 'sex', 'age', 'suicides_no', 'population',
       'suicides/100k pop', 'gdp_for_year ($)', 'gdp_per_capita ($)',
       'generation', 'new_age', 'sex_numeric', 'gdp_billion'],
      dtype='str')

In [52]:
# Is there a significant difference in suicide rates between males and females? 
gender_deff=df.groupby('sex')['suicides/100k pop'].mean().reset_index()
gender_deff

,sex,suicides/100k pop
0,female,5.392866
1,male,20.239329


In [53]:
px.pie(gender_deff,names='sex',values='suicides/100k pop')

- Finding: There is a significant gender gap in suicide rates. On average, the suicide rate for males (approx. 20.24 per 100k) is nearly four times higher than that for females (approx. 5.39 per 100k). This suggests that males are a much higher-risk group globally.

In [58]:
comparison = df.groupby(['sex', 'new_age','age'])['suicides/100k pop'].mean().reset_index()
comparison

,sex,new_age,age,suicides/100k pop
0,female,0,5-14 years,0.462252
1,female,1,15-24 years,4.328018
2,female,2,25-34 years,4.573210
3,female,3,35-54 years,5.916480
4,female,4,55-74 years,7.123447
5,female,5,75+ years,9.919802
6,male,0,5-14 years,0.777831
7,male,1,15-24 years,13.566290
8,male,2,25-34 years,19.800560
9,male,3,35-54 years,23.978402


In [60]:
px.line(comparison, x='age', y='suicides/100k pop', color='sex',title='Suicide Rates by Age Group')

- Finding: Suicide rates show a strong positive correlation with age. Both males and females experience an increase in suicide rates as they get older, with the 75+ years age group consistently recording the highest rates.

In [ ]:
# Has this gap closed or widened over the years?

gap_trend = df.groupby(['year', 'sex'])['suicides/100k pop'].mean().reset_index()

## Because the Data of 2016 is not complete (Sample Size Bias)
gap_trend_clean = gap_trend[gap_trend['year'] < 2016]

px.line(gap_trend_clean, x='year', y='suicides/100k pop', color='sex', 
        title='Gender Gap Trend (1985 - 2015) - Excluding Incomplete 2016 Data')

- Analysis Conclusion: > The gender gap widened significantly between 1985 and 1995, as male suicide rates surged while female rates remained relatively stable. Although there was a gradual decline for both genders after 1995, the gap remains substantial, with male rates consistently staying 3 to 4 times higher than female rates. A sharp increase is also observed in 2016, which requires further investigation to see if it marks a new rising trend.



In [64]:
# Generational Trends: Which generation shows the highest suicide risk?
generation_trend = df.groupby(['generation', 'sex'])['suicides/100k pop'].mean().reset_index()
px.bar(generation_trend, x='generation', y='suicides/100k pop', color='sex', title='Suicide Rates by Generation')


- Finding: Suicide risk is strongly associated with older generations. The G.I. Generation and Silent Generation exhibit the highest suicide rates globally. There is a clear downward trend as we move toward younger generations (Millennials and Gen Z). This reinforces our previous finding that suicide rates increase with age.

In [65]:
# Global Trajectory: How has the total number of suicides evolved from 1985 to 2016? Is the global trend increasing or decreasing?
global_trend = df[df['year'] < 2016].groupby('year')['suicides_no'].sum().reset_index()

px.line(global_trend, x='year', y='suicides_no', title='Global Trajectory of Suicides (Total Numbers: 1985 - 2015)',markers=True)

- 1985-1999: A significant upward trend, with global suicide counts nearly doubling.

Peak Period: The numbers reached their maximum between 1999 and 2003.

Current Trend: Since 2003, there has been a notable decrease in the total number of suicides globally, returning to levels near those of the early 1990s. This overall trajectory suggests that while the issue peaked at the turn of the millennium, global efforts or social changes have successfully started to reduce the numbers in the last decade.

In [75]:
# Which 10 countries have the highest total suicides_no, and how does this list change when looking at suicides/100k pop?
top_10_countries = df.groupby('country')['suicides_no'].sum().sort_values(ascending=False).head(10).reset_index()
top_10_countries_by_rate = df.groupby('country')['suicides/100k pop'].mean().sort_values(ascending=False).head(10).reset_index()
print(top_10_countries)
print('-'*100)
print(top_10_countries_by_rate)

              country  suicides_no
0  Russian Federation      1209742
1       United States      1034013
2               Japan       806902
3              France       329127
4             Ukraine       319950
5             Germany       291262
6   Republic of Korea       261730
7              Brazil       226613
8              Poland       139098
9      United Kingdom       136805
----------------------------------------------------------------------------------------------------
              country  suicides/100k pop
0           Lithuania          40.415573
1           Sri Lanka          35.295152
2  Russian Federation          34.892377
3             Hungary          32.761516
4             Belarus          31.075913
5          Kazakhstan          30.511282
6              Latvia          29.259325
7            Slovenia          27.827857
8             Estonia          27.276905
9             Ukraine          26.582321


In [ ]:
from plotly.subplots import make_subplots
import plotly.graph_objects as go

fig = make_subplots(rows=1, cols=2, subplot_titles=("Top 10 by Total Suicides", "Top 10 by Rate (per 100k)"),horizontal_spacing=0.15)

fig.add_trace(go.Bar(x=top_10_countries['country'], y=top_10_countries['suicides_no'], name="Total"),row=1, col=1)

fig.add_trace(go.Bar(x=top_10_countries_by_rate['country'], y=top_10_countries_by_rate['suicides/100k pop'], name="Rate"),row=1, col=2)

fig.update_layout(height=500, width=1200, title_text="Geographic Comparison: Totals vs. Rates", showlegend=False)
fig.show()

- Geographic Insight:
While the Russian Federation, USA, and Japan record the highest total number of suicides, the picture changes drastically when looking at the suicide rate per 100k population. Lithuania emerges as the highest-risk country globally, followed by Sri Lanka. Interestingly, Russia is the only country that appears in the top 3 of both lists, indicating it faces both a high total burden and a high individual risk. This confirms that total counts are often biased by population size, making rates a more accurate metric for comparing national risk levels.

In [83]:
#Do countries with similar cultures or geographic locations (e.g., Nordic countries or Middle Eastern countries) share similar suicide rate patterns?
middle_east_list = ['Bahrain', 'Kuwait', 'Oman', 'Qatar', 'United Arab Emirates', 'Turkey']
# feature Engineering 
df['Category'] = 'Rest of the World' 
df.loc[df['country'].isin(middle_east_list), 'Category'] = 'Middle East' 
comparison = df.groupby(['year', 'Category'])['suicides/100k pop'].mean().reset_index()

px.line(comparison, x='year', y='suicides/100k pop', color='Category',
        title='The Cultural Impact: Middle East vs. Global Average',
        line_shape='spline', 
        markers=True)

- Key Finding:
There is a profound and consistent disparity between the Middle East and the Rest of the World. The Middle East maintains a significantly lower suicide rate (consistently below 3 per 100k), showcasing a remarkable level of cultural resilience. While the global average experienced sharp increases in the 1990s due to socio-economic crises, the Middle Eastern trend remained relatively flat and stable. This suggests that strong social, religious, and family-oriented structures in the region act as a powerful deterrent against suicide, regardless of global external shifts.

In [87]:
#Wealth Correlation: Does a higher gdp_per_capita (individual wealth) correlate with lower suicide rates, or is the opposite true?

gdp_analysis = df.groupby('country').agg({'suicides/100k pop': 'mean','gdp_per_capita ($)': 'mean'}).reset_index()

fig = px.scatter(gdp_analysis,x='gdp_per_capita ($)', y='suicides/100k pop',hover_name='country',trendline="ols", title='Correlation: Individual Wealth vs. Suicide Rates')

fig.show()


- Critical Finding:
The correlation between GDP per capita and suicide rates is remarkably weak/neutral. The trend line is almost flat, indicating that individual wealth is NOT a primary predictor of suicide risk. We see high-income countries and low-income countries sharing similar rates. This suggests that socio-economic status alone cannot explain the phenomenon; instead, cultural, psychological, and social support factors likely play a more dominant role.

In [92]:
korea_data = df[df['country'] == 'Republic of Korea'].groupby('year').agg({
    'suicides/100k pop': 'mean',
    'gdp_per_capita ($)': 'mean'
}).reset_index()

fig = make_subplots(rows=2, cols=1, 
                    subplot_titles=("Growth of Wealth (GDP per Capita)", "Growth of Suicide Rates"),
                    vertical_spacing=0.15)

fig.add_trace(
    go.Scatter(x=korea_data['year'], y=korea_data['gdp_per_capita ($)'], name="Wealth", line=dict(color='green')),
    row=1, col=1
)

fig.add_trace(
    go.Scatter(x=korea_data['year'], y=korea_data['suicides/100k pop'], name="Suicide Rate", line=dict(color='red')),
    row=2, col=1
)

fig.update_layout(height=700, title_text="The South Korean Paradox: Wealth vs. Mental Health", showlegend=False)
fig.show()

- Case Study Conclusion (South Korea):
South Korea presents a unique paradox. While the GDP per capita grew exponentially (from $3k to $30k), the suicide rate followed the same upward trajectory, peaking at over 40 per 100k. This directly answers the "Economic Transition" question: Rapid wealth accumulation does not always correlate with better mental health. In fact, it suggests that intense economic competition and social shifts during rapid development can act as significant stressors, leading to higher suicide rates.



In [93]:
df.columns

Index(['country', 'year', 'sex', 'age', 'suicides_no', 'population',
       'suicides/100k pop', 'gdp_for_year ($)', 'gdp_per_capita ($)',
       'generation', 'new_age', 'sex_numeric', 'gdp_billion', 'region',
       'Category'],
      dtype='str')